server/filestore.go

```diff
@@ -8382,7 +8382,13 @@ func (fs *fileStore) Snapshot(deadline time.Duration, checkMsgs, includeConsumer
 	}
 
 	// Write out full state as well before proceeding.
-	fs.writeFullState()
```

Comment:
My sense is we want this written at the end not beginning. Will be looking at this as we discussed in Slack.

Thanks! But ok to close for now..